staging/src/k8s.io/apimachinery/pkg/util/diff/diff.go

```diff
@@ -50,220 +46,44 @@ func StringDiff(a, b string) string {
 	return string(out)
 }
 
-// ObjectDiff writes the two objects out as JSON and prints out the identical part of
-// the objects followed by the remaining part of 'a' and finally the remaining part of 'b'.
-// For debugging tests.
-func ObjectDiff(a, b interface{}) string {
-	ab, err := json.Marshal(a)
-	if err != nil {
-		panic(fmt.Sprintf("a: %v", err))
-	}
-	bb, err := json.Marshal(b)
-	if err != nil {
-		panic(fmt.Sprintf("b: %v", err))
-	}
-	return StringDiff(string(ab), string(bb))
-}
-
-// ObjectGoPrintDiff is like ObjectDiff, but uses go-spew to print the objects,
-// which shows absolutely everything by recursing into every single pointer
-// (go's %#v formatters OTOH stop at a certain point). This is needed when you
-// can't figure out why reflect.DeepEqual is returning false and nothing is
-// showing you differences. This will.
-func ObjectGoPrintDiff(a, b interface{}) string {
-	s := spew.ConfigState{DisableMethods: true}
-	return StringDiff(
-		s.Sprintf("%#v", a),
-		s.Sprintf("%#v", b),
-	)
-}
-
-func ObjectReflectDiff(a, b interface{}) string {
-	vA, vB := reflect.ValueOf(a), reflect.ValueOf(b)
-	if vA.Type() != vB.Type() {
-		return fmt.Sprintf("type A %T and type B %T do not match", a, b)
-	}
-	diffs := objectReflectDiff(field.NewPath("object"), vA, vB)
-	if len(diffs) == 0 {
-		return "<no diffs>"
-	}
-	out := []string{""}
-	for _, d := range diffs {
-		elidedA, elidedB := limit(d.a, d.b, 80)
-		out = append(out,
-			fmt.Sprintf("%s:", d.path),
-			fmt.Sprintf("  a: %s", elidedA),
-			fmt.Sprintf("  b: %s", elidedB),
-		)
-	}
-	return strings.Join(out, "\n")
+// Testing is an interface that *testing.T implements.
+type Testing interface {
```

Comment:
I think that I want to copy this until I can set the package to testonly. Otherwise I can't guarantee that this library won't leak test flags into binaries. There are already bad deps on the diff package.